geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/PersistentBucketRecoverer.java

```diff
@@ -246,8 +264,12 @@ public RegionStatus(PartitionedRegion region) {
     }
 
     public void removeListeners() {
+      BucketPersistenceAdvisor tempBPA;
       for (ProxyBucketRegion proxyBucket : bucketRegions) {
-        proxyBucket.getPersistenceAdvisor().removeListener(PersistentBucketRecoverer.this);
+        tempBPA = proxyBucket.getPersistenceAdvisor();
+        if (!tempBPA.isClosed()) {
+          proxyBucket.getPersistenceAdvisor().removeListener(PersistentBucketRecoverer.this);
```

Comment:
why do you refetch "proxyBucket.getPersistenceAdvisor()" instead of using tempBPA?
Also, would it be better to do this isClosed check in BucketPersistenceAdvisor?
Also why did you make this particular change? You have changed close to set the listeners to an empty collection. You could just have removeListener check if the set isEmpty and return. It seems like this method would not need to change at all

Reply:
Thanks for comments. Updated